src/components/Footer.tsx

```diff
@@ -0,0 +1,114 @@
+import { Link, LinkProps, Typography, useTheme } from "@mui/material";
+import Grid from "@mui/material/Grid2";
+import dlsLogo from "../public/dls.svg";
+
+interface FooterLinksProps {
+  children: React.ReactElement<LinkProps> | React.ReactElement<LinkProps>[];
+}
+
+interface FooterProps {
+  /** Location/content of the logo */
+  logo?: string | null;
+  copyright?: string | null;
+  children?: React.ReactElement | React.ReactElement[];
+}
+
+const FooterLinks = ({ children }: FooterLinksProps) => {
+  return (
+    <div
+      style={{
+        float: "left",
+        alignItems: "center",
+        borderTop: "4px solid transparent",
+        borderBottom: "4px solid transparent",
+        display: "flex",
+        flexWrap: "wrap",
+      }}
+    >
+      {children}
+    </div>
+  );
+};
+
+const FooterLink = ({ children, ...props }: LinkProps) => {
+  const theme = useTheme();
+
+  return (
+    <Link
+      sx={{
+        "&:hover": {
+          color: theme.palette.secondary.main,
+          borderBottom: "solid 4px",
+        },
+        textDecoration: "none",
+        color: theme.palette.primary.contrastText,
+        marginLeft: "1.5rem",
+        cursor: "pointer",
+      }}
+      {...props}
+    >
+      {children}
+    </Link>
+  );
+};
+
+/*
+ * Basic footer bar.
+ * Can be used with `FooterLinks` and `FooterLink` to display a list of links.
+ */
+const Footer = ({
+  logo = dlsLogo as string,
+  copyright,
+  children,
+  ...props
+}: FooterProps) => {
+  const theme = useTheme();
+
+  return (
+    <footer
+      style={{
+        bottom: 0,
+        marginTop: "auto",
+        minHeight: 50,
+        backgroundColor: theme.palette.primary.light,
+      }}
+      {...props}
+    >
+      <Grid container>
+        <Grid
+          size={{ xs: 6, md: 8 }}
+          style={{
+            alignContent: "center",
+          }}
+        >
+          {children}
+        </Grid>
+        <Grid size={{ xs: 6, md: 4 }}>
+          <div
+            style={{
+              float: "right",
+              paddingTop: "10px",
+              paddingRight: "15px",
+              textAlign: "right",
+            }}
+          >
+            {logo ? <img alt="footer-logo" src={logo} /> : null}
+            {copyright ? (
+              <Typography
+                style={{
+                  margin: 0,
+                  color: theme.palette.primary.contrastText,
+                }}
+              >
+                {copyright} {new Date().getFullYear()}
```

Comment:
The new version isn't quite right. There's an established way to write a copywrite and the date must go immediately after the copyright icon.

Let's make this easier. Well just allow the last part to be configurable:
```suggestion
                {"Copyright © " + new Date().getFullYear() + copyright}
```

When you switch over the Storybook versions can you put "Company" in instead of just "text" (or "Acme" or something similar!)

Reply:
![Screenshot From 2024-12-12 13-36-04](https://github.com/user-attachments/assets/0586e407-3751-45fc-b924-813d9ef3026c)